tutorials/EmbeddingObjectives.qmd

```diff
@@ -0,0 +1,189 @@
+for the background on converting Hessians.
+
+Here we use the Examples 9.40 and 9.49 of [Boumal:2023](@cite) and compare the different methods,
+one can call the solver, depending on which gradient and/or Hessian one provides.
+
+```{julia}
+#| echo: false
+#| code-fold: true
+#| output: false
+# For now use the local one since Manifolds.jl #644 is required for this
+using Pkg;
+cd(@__DIR__)
+Pkg.activate("."); # for reproducibility use the local tutorial environment.
+```
+
+```{julia}
+#| output: false
+using Manifolds, Manopt, ManifoldDiff
+using LinearAlgebra, Random, Colors, Plots
+Random.seed!(123)
+```
+
+We consider the cost function on the [`Grassmann`](https://juliamanifolds.github.io/Manifolds.jl/latest/manifolds/grassmann.html) manifold given by
+
+```{julia}
+n = 5
+k = 2
+M = Grassmann(5,2)
+A = Symmetric(rand(n,n));
+```
+
+```{julia}
+#| output: false
+f(M, p) = 1 / 2 * tr(p' * A * p)
+```
+
+Note that this implementation is already also a valid implementation / continuation
+of $f$ into the (lifted) embedding of the Grassmann manifold.
+In the implementation we can use `f` for both the Euclidean $\tilde f$ and the Grassmann case $f$.
+
+Its Euclidean gradient and Hessian are easy to compute as
+
+```{julia}
+#| output: false
+∇f(M, p) = A * p
+∇²f(M,p,X) = A*X
+```
+
+On the other hand, from the aforementioned Example 9.49 we can also state
+the Riemannian gradient and Hessian for comparison as
+
+```{julia}
+#| output: false
+grad_f(M, p) = A * p - p * (p' * A * p)
+Hess_f(M, p, X) = A * X - p * p' * A * X - X * p' * A * p
+```
+
+We can check that these are the correct at least numerically by calling
+the [`check_gradient`](@ref)
+
+```{julia}
+check_gradient(M, f, grad_f; plot=true)
+```
+
+and the [`check_Hessian`](@ref), which requires a bit more tolerance in its linearity check
+
+```{julia}
+check_Hessian(M, f, grad_f, Hess_f; plot=true, throw_error=true, atol=1e-15)
+```
+
+While they look reasonable here and were already derived – for the general case this derivation
+might be more complicated.
+
+Luckily there exist two functions in [`ManifoldDiff.jl`](https://juliamanifolds.github.io/ManifoldDiff.jl/stable/) that are implemented for several
+manifolds from [`Manifolds.jl`](https://github.com/JuliaManifolds/Manifolds.jl), namely [`riemannian_gradient`](https://juliamanifolds.github.io/ManifoldDiff.jl/stable/library/#ManifoldDiff.riemannian_gradient-Tuple{AbstractManifold,%20Any,%20Any})`(M, p, eG)` that converts a Riemannian gradient
+`eG=`$\nabla \tilde f(p)$  into a the Riemannain one $\operatorname{grad} f(p)$
+ and [`riemannian_Hessian`](https://juliamanifolds.github.io/ManifoldDiff.jl/stable/library/#ManifoldDiff.riemannian_Hessian-Tuple{AbstractManifold,%20Any,%20Any,%20Any,%20Any})`(M, p, eG, eH, X)
+ which converts the Euclidean Hessian `eH=`$\nabla^2 \tilde f(p)[X]$  into $\operatorname{Hess} f(p)[X]$,
+ where we also require the Euclidean gradient `eG=`$\nabla \tilde f(p)$.
```

Comment:
There are some formatting errors around this part.

Reply:
Thanks for spotting, for now I am still fighting interactions of sub solvers and embeddings ;)